private/endpoints/rulesfn/error_collector.go

```diff
@@ -0,0 +1,49 @@
+package rulesfn
+
+import "strings"
+
+// ErrorCollector provides collection of errors that occurred while evaluating
+// endpoint rule functions.
+type ErrorCollector struct {
```

Comment:
**Question**: I still don't understand who is the consumer of errors collected by this abstraction? How do we see this playing out in practice?

Reply:
@aajtodd you can see a full example [here](https://github.com/aws/smithy-go/pull/388) if you click Example generated output in one of the comments. However, per standup discussion, im removing for now and will revisit later if i start to feel strongly about it.